### ISSUE TITLE
Missing git submodule init - extern_iowow does not appear to contain CMakeLists.txt

### ISSUE THREAD

**Issue by abs0** [Softmotions/ejdb#341]
Versions:

```
% sw_vers
ProductName:	macOS
ProductVersion:	12.2.1
BuildVersion:	21D62
% cmake --version
cmake version 3.20.5

CMake suite maintained and supported by Kitware (kitware.com/cmake).

% yarn --version
1.22.15
```

Attempt to build:

```
% git clone https://github.com/Softmotions/ejdb.git

Cloning into 'ejdb'...
remote: Enumerating objects: 20586, done.
remote: Counting objects: 100% (1444/1444), done.
remote: Compressing objects: 100% (994/994), done.
remote: Total 20586 (delta 885), reused 822 (delta 346), pack-reused 19142
Receiving objects: 100% (20586/20586), 10.99 MiB | 7.29 MiB/s, done.
Resolving deltas: 100% (13294/13294), done.
% cd ejdb
% mkdir build && cd build
% cmake .. -DCMAKE_BUILD_TYPE=Release
-- Using project changelog file: /Users/abs/ejdb/Changelog
-- The C compiler identification is AppleClang 13.0.0.13000029
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Resolving GIT Version
-- Found Git: /usr/bin/git (found version "2.32.0 (Apple Git-132)") 
-- GIT hash: 7c7fe884
IOWOW_URL: /Users/abs/ejdb/extra/iowow
IOWOW CMAKE_ARGS: -DOWNER_PROJECT_NAME=ejdb2;-DCMAKE_BUILD_TYPE=Release;-DCMAKE_INSTALL_PREFIX=/Users/abs/ejdb/build;-DASAN=OFF;-DBUILD_SHARED_LIBS=OFF;-DBUILD_EXAMPLES=OFF;-DCMAKE_OSX_ARCHITECTURES=;-DCMAKE_C_COMPILER=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
IWNET_URL: /Users/abs/ejdb/extra/iwnet
IWNET CMAKE_ARGS: -DOWNER_PROJECT_NAME=ejdb2;-DCMAKE_BUILD_TYPE=Release;-DCMAKE_INSTALL_PREFIX=/Users/abs/ejdb/build;-DASAN=OFF;-DBUILD_SHARED_LIBS=OFF;-DBUILD_EXAMPLES=OFF;-DCMAKE_FIND_ROOT_PATH=/usr/local;-DCMAKE_OSX_ARCHITECTURES=;-DCMAKE_C_COMPILER=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  

CMAKE_GENERATOR: Unix Makefiles
ejdb2 LINK LIBS: IWNET::static;IOWOW::static;m
ejdb2 ASAN: OFF

ejdb2 INCLUDE DIRS: /Users/abs/ejdb/build/src/generated;/Users/abs/ejdb/src;/Users/abs/ejdb/build/include;/Users/abs/ejdb/build/include/ejdb2;/Users/abs/ejdb/src/util;/Users/abs/ejdb/src/jbl;/Users/abs/ejdb/src/jql;/Users/abs/ejdb/src/jbi;/Users/abs/ejdb/src/jbr

ejdb2 SOURCES: /Users/abs/ejdb/src/ejdb2.c;/Users/abs/ejdb/src/util/utf8proc.c;/Users/abs/ejdb/src/jbl/binn.c;/Users/abs/ejdb/src/jbl/jbl.c;/Users/abs/ejdb/src/jbl/jbl_json.c;/Users/abs/ejdb/src/jql/jql.c;/Users/abs/ejdb/src/jql/jqp.c;/Users/abs/ejdb/src/jbi/jbi_consumer.c;/Users/abs/ejdb/src/jbi/jbi_dup_scanner.c;/Users/abs/ejdb/src/jbi/jbi_full_scanner.c;/Users/abs/ejdb/src/jbi/jbi_pk_scanner.c;/Users/abs/ejdb/src/jbi/jbi_selection.c;/Users/abs/ejdb/src/jbi/jbi_sorter_consumer.c;/Users/abs/ejdb/src/jbi/jbi_uniq_scanner.c;/Users/abs/ejdb/src/jbi/jbi_util.c;/Users/abs/ejdb/src/jbr/jbr.c

ejdb2 PUB_HDRS: /Users/abs/ejdb/src/ejdb2.h;/Users/abs/ejdb/src/jbl/jbl.h;/Users/abs/ejdb/src/jbr/jbr.h;/Users/abs/ejdb/src/jql/jql.h

ejdb2 CMAKE_BUILD_TYPE: Release
ejdb2 ANDROID_ABIS: x86;x86_64;arm64-v8a;armeabi-v7a
ejdb2 ENABLE_HTTP: ON
ejdb2 BUILD_SHARED_LIBS: ON
ejdb2 BUILD_TESTS: OFF
ejdb2 BUILD_EXAMPLES: ON
ejdb2 BUILD_BENCHMARKS: OFF
ejdb2 BUILD_DART_BINDING: OFF
ejdb2 BUILD_ANDROID_LIBS: OFF
ejdb2 BUILD_JNI_BINDING: OFF
ejdb2 BUILD_NODEJS_BINDING: OFF
ejdb2 BUILD_REACT_NATIVE_BINDING: OFF
ejdb2 BUILD_FLUTTER_BINDING: OFF
ejdb2 BUILD_SWIFT_BINDING: OFF
ejdb2 GIT_REVISION: 7c7fe884
ejdb2 CMAKE_INSTALL_PREFIX: /usr/local
ejdb2 CPACK_GENERATORS: TGZ;ZIP
ejdb2 CMAKE_GENERATOR: Unix Makefiles
ejdb2 CMAKE_SYSTEM_NAME: Darwin
ejdb2 CPU: x86_64
ejdb2 SIZEOF *VOID: 8
ejdb2 PROJECT: ejdb2 (2.72) testing; urgency=medium
ejdb2 CHANGELOG_MESSAGE:
  * Upgraded to the stable verion of iwnet lib.
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/abs/ejdb/build
% make install
[  1%] Creating directories for 'extern_iowow'
[  3%] Performing download step (DIR copy) for 'extern_iowow'
[  5%] No patch step for 'extern_iowow'
[  7%] Performing configure step for 'extern_iowow'
CMake Error: The source directory "/Users/abs/ejdb/build/src/extern_iowow" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
make[2]: *** [src/extern_iowow-stamp/extern_iowow-configure] Error 1
make[1]: *** [src/CMakeFiles/extern_iowow.dir/all] Error 2
make: *** [all] Error 2

```

**Comment by abs0** [comment]
It looks like this may have been introduced with 0b7f70a3793ddfb58df483a3a6132a8d6574e839 (Migration to iwnet) though the error in that commit shows up extern_wslay

```
-- Build files have been written to: /Users/abs/ejdb/build
[  1%] Creating directories for 'extern_wslay'
[  2%] Performing download step (DIR copy) for 'extern_wslay'
[  4%] No patch step for 'extern_wslay'
[  5%] Performing configure step for 'extern_wslay'
CMake Error: The source directory "/Users/abs/ejdb/build/src/extern_wslay" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.

```

**Comment by abs0** [comment]
It looks like it might just be a doc issue - if "git submodule init && git submodule update" are run it seems to build. Not quite sure how this will affect the embedded flutter build, but that would be the next thing for me to investigate...

**Comment by abs0** [comment]
Would it be difficult to adjust the build to detect the empty iowow directory and fail with a helpful message?

**Comment by abs0** [comment]
For my specific use case (building EJDB2 into a flutter app), resolved by Softmotions/EJDB2Swift#1
